Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EditorEntitySortComponentSerializer.cpp

```diff
@@ -122,10 +123,31 @@ namespace AzToolsFramework::Components
             const EntityOrderArray* defaultChildEntityOrderArray =
                 defaultsortComponentInstance ? &defaultsortComponentInstance->m_childEntityOrderArray : nullptr;
 
+            // Sort order array may contain stale entityIds that can be ignored. This is because
+            // adding/removing a child entity also affects its parent entity's sort order component.
+            // So when an entity is added/removed as a prefab override, two separate overrides are
+            // generated, one referencing the child and one referencing the parent.
+            // Reverting prefab overrides only affects one entity at a time, so reverting overrides on
+            // a parent whose child was deleted as an override would cause the deleted child's entityId
+            // to be added back to the parent's sort order array
```

Comment:
nit: Instead of giving a very specific repro case here, it's better to keep it general since the warning was previously being sent anytime it can't resolve an entity id reference. 
Something like "Since sort order component on an entity stores a vector of child entity ids in it, it's possible for the component to be left with stale entity ids when updates on child entities don't update the parent entity's sort order component. To reduce the noise in error log, we are   allowing the component to have stale entity ids."